File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
     
    int n = nums.size();
    std::vector<int> result(n, 1);

    int prefix  = 1;
    for (int i = 0; i < n; i++) {
        result[i] *= prefix ;
        prefix  *= nums[i];
    }

     int suffix  = 1;
    for (int i = n - 1; i >= 0; i--) {
        result[i] *= suffix ;
        suffix  *= nums[i];
    }

    return result;  
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    const n = nums.length
    const result = Array.from({ length: n}, () => 1);
    let prefix = 1;
    for(let i = 0; i < n;i++)
    {
        result[i] *= prefix;
        prefix *= nums[i]
    }
    let suffix = 1;
    for(let i = n - 1; i >= 0;i--)
    {
        result[i] *= suffix;
        suffix *= nums[i]
    }
    return result;
    
};